assembly/src/parsers/io_ops.rs

```diff
@@ -256,69 +276,137 @@ pub fn parse_loc_storew(op: &Token) -> Result<Node, ParsingError> {
-    let mut constants = Vec::new();
-    let param_idx = 1;
-    let param_count = op.num_parts() - param_idx;
+/// Parses a hexadecimal or decimal parameter values into a vector of u64 values valid for field
+/// element creation and returns appropriate instruction based on parsed values
+fn parse_param_list(op: &Token) -> Result<Node, ParsingError> {
+    let mut values = Vec::with_capacity(op.num_parts() - 1);
+    for (param_idx, &param_str) in op.parts().iter().enumerate().skip(1) {
+        let value = match param_str.strip_prefix("0x") {
+            Some(param_str) => parse_hex_value(op, param_str, param_idx)?,
+            None => parse_checked_param(op, param_idx, 0, Felt::MODULUS - 1)?,
+        };
+        values.push(value);
+    }
 
-    // for multiple input parameters, parse & push each one onto the stack in order, then return
-    if param_count > 1 {
-        for param_idx in param_idx..=param_count {
-            let value = parse_element_param(op, param_idx)?;
-            constants.push(value);
-        }
-        return Ok(constants);
+    build_push_many_instruction(values.iter())
+}
+
+/// Parses a hexadecimal parameter values in the form of strings of lenght 16 without delimeters
+/// into a vector of u64 values and returns appropriate instruction based on parsed values
+///
+/// # Errors
+/// Returns an error if:
+/// - The length of hex string is not even.
+/// - The length of hex string is not divisible by 16.
+/// - If the string does not contain a valid hexadecimal value.
+/// - If the parsed value is greater than or equal to the field modulus.
+fn parse_long_hex_param(op: &Token, param_str: &str) -> Result<Node, ParsingError> {
+    // handle error cases where the hex string is poorly formed
+    if param_str.len() % HEX_CHUNK_SIZE != 0 {
+        // hex string doesn't contain a valid number of bytes
+        return Err(ParsingError::invalid_param_with_reason(
+            op,
+            1,
+            &format!(
+                "hex string '{param_str}' does not contain a number of characters multiple 16"
+            ),
+        ));
     }
 
-    // for a single input, there could be one value or there could be a series of many hexadecimal
-    // values without separators
-    let param_str = op.parts()[param_idx];
-    if let Some(param_str) = param_str.strip_prefix("0x") {
-        // parse 1 or more hexadecimal values
-        let values = parse_hex_params(op, param_idx, param_str)?;
-        // push each value onto the stack in order
-        for &value in values.iter() {
-            constants.push(value);
-        }
+    // iterate over the multi-value hex string and parse each 8-byte chunk into a valid u64
+    let values = (0..param_str.len())
+        .step_by(HEX_CHUNK_SIZE)
+        .map(|i| parse_hex_value(op, &param_str[i..i + HEX_CHUNK_SIZE], 1))
+        .collect::<Result<Vec<_>, _>>()?;
+
+    build_push_many_instruction(values.iter())
```

Comment:
You don't need to do `collect()` here. You can just pas the iterator directly to `build_push_many_instructions()`.

You can probably do the same thing for `parse_param_list()`. Though there you'd need to transform the `for` loop into an iterator.

Reply:
`.map()` here produces a `Map<>` type which I can't cast to `Iter`. @vlopes11 do you know any way to do that?

Reply:
All related solutions that I found use `.collect()` after map, I didn't find any cases where returning values from `.map()` has been used as a plain iterator over map values, like in our case. I don't think that `.map()` returning values could be used like so, even though it is an iterator

Reply:
@vlopes11 - do you have any suggestions here?

Reply:
The problem was in `build_push_many_instructions`. I suggested one change there that will allow us to not collect